frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -248,17 +311,38 @@ export const parseEncodedSnapshots = async (
                 isMobileSnapshots = hasAnyWireframes(snapshotData)
             }
 
-            return snapshotData.map((d: unknown) => {
+            return snapshotData.flatMap((d: unknown, snapshotIndex) => {
                 const snap = coerceToEventWithTime(d, withMobileTransformer)
 
-                return {
-                    // this handles parsing data that was loaded from blob storage "window_id"
-                    // and data that was exported from the front-end "windowId"
-                    // we have more than one format of data that we store/pass around
-                    // but only one that we play back
+                const baseSnapshot: RecordingSnapshot = {
                     windowId: snapshotLine['window_id'] || snapshotLine['windowId'],
                     ...snap,
                 }
+
+                // Find the next snapshot if possible
+                let nextSnapshot: RecordingSnapshot | null = null
+                if (index < items.length - 1 && snapshotIndex === snapshotData.length - 1) {
+                    try {
+                        const nextItem = items[index + 1]
+                        if (typeof nextItem === 'string') {
+                            const parsed = JSON.parse(nextItem)
+                            const nextData = parsed['data']?.[0]
+                            if (nextData) {
+                                nextSnapshot = coerceToEventWithTime(
+                                    nextData,
+                                    withMobileTransformer
+                                ) as RecordingSnapshot
+                            }
+                        } else if (isRecordingSnapshot(nextItem)) {
+                            nextSnapshot = nextItem
+                        }
```

Comment:
i wonder if this would be simpler if we reversed the list first...
so we always have the "next" snapshot because it was the last one we processed

so...

reverse the list

for each item
   * coerce to eventwithtime
   * check if it needs chunking
   * store its timestamp
 
 then reverse the list again

Reply:
(I've no idea the performance of reversing the list twice if the data is very large, but then we have `...snap` already which is going to be nasty with big data already so 🤷 )